pkg/kv/kvserver/replica_application_result.go

```diff
@@ -407,6 +407,8 @@ func (r *Replica) tryReproposeWithNewLeaseIndex(ctx context.Context, origCmd *re
 		// The tracker wants us to forward the request timestamp, but we can't
 		// do that without re-evaluating, so give up. The error returned here
 		// will go to back to DistSender, so send something it can digest.
+		r.mu.RLock()
+		defer r.mu.RUnlock()
```

Comment:
Took me a few seconds to realize this defer won't leave this block because of the return.
Optionally inline the return as `return kvpb.NewNotLeaseHolderError(...)`, to make it visually closer to the defer.